jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java

```diff
@@ -0,0 +1,427 @@
+                fieldToRemove = field;
+                valueToRemove = (String)entity.get(field);
+                break;
+            }
+        }
+        //First Search of the field exists or not
+        SearchResponse existsResponse = client.prepareSearch(INDEX_NAME)
+                .setTypes(docDef.getEntityField())
+                .setQuery(QueryBuilders.existsQuery(fieldToRemove))
+                .get();
+
+        String script = null;
+        if(existsResponse != null && existsResponse.getHits() != null && existsResponse.getHits().totalHits() > 0) {
+            //This means field already exists and therefore we should remove from it
+            script = "ctx._source." + fieldToRemove+".remove('"+ valueToRemove + "')";
+        }
+
+        UpdateRequest updateRequest = new UpdateRequest(INDEX_NAME, docDef.getEntityField(), entity.getId())
+                .script(new Script(script));
+
+        try {
+            client.update(updateRequest).get();
+        }catch(Exception e) {
+            throw new TextIndexException("Unable to delete entity.", e);
+        }
+
+
+        LOGGER.debug("deleting content related to entity: " + entity.getId());
+//        client.prepareDelete(INDEX_NAME, docDef.getEntityField(), entity.getId()).get();
+
+    }
+
+    /**
+     * Get an Entity given the subject Id
+     * @param uri the subject Id of the entity
+     * @return a map of field name and field values;
+     */
+    @Override
+    public Map<String, Node> get(String uri) {
+
+        GetResponse response;
+        Map<String, Node> result = new HashMap<>();
+
+        if(uri != null) {
+            response = client.prepareGet(INDEX_NAME, docDef.getEntityField(), uri).get();
+            if(response != null && !response.isSourceEmpty()) {
+                String entityField = response.getId();
+                Node entity = NodeFactory.createURI(entityField) ;
+                result.put(docDef.getEntityField(), entity);
+                for (String field: docDef.fields()) {
+
+                    GetField fieldResponse = response.getField(field);
+
+                    if(fieldResponse == null || fieldResponse.getValue() == null) {
+                        //We wont return it.
+                        continue;
+                    }
+                    if(fieldResponse instanceof List<?>) {
+                        //We are only interested in literal values
+                        continue;
+                    }
+                    //We assume it will always be a String value.
+                    String fieldValue = (String)fieldResponse.getValue();
+                    Node fieldNode = NodeFactoryExtra.createLiteralNode(fieldValue, null, null);
+                    result.put(field, fieldNode);
+
+                }
+
+
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    public List<TextHit> query(Node property, String qs) {
+
+        return query(property, qs, 0);
```

Comment:
The query limit defaults to 0. I think this will give zero results though I'm not sure if ES considers the value 0 special. jena-text/Lucene has a constant MAX_N (defined as 10000) which is used as the default in this situation.